src/Registration/cReg/include/sirf/cReg/SIRFRegParserKey.h

```diff
@@ -0,0 +1,266 @@
+
+    /// Get argument - float
+    void get_argument(std::string line, int arg_num, float &arg        ) { arg = std::stof(get_arg_as_string(line, arg_num));  }
+
+    /// Get argument - double
+    void get_argument(std::string line, int arg_num, double &arg       ) { arg = std::stod(get_arg_as_string(line, arg_num));  }
+
+    /// Get argument - unsigned long
+    void get_argument(std::string line, int arg_num, unsigned long &arg) { arg = std::stoul(get_arg_as_string(line, arg_num)); }
+
+    /// Object to call the function on
+    std::shared_ptr<Z> _object;
+};
+
+/// Abstract base class for a parser key with 0 arguments
+template<class Z>
+class SIRFRegParserKey0Arg : public SIRFRegParserKeyBase<Z>
+{
+public:
+
+    /// Set function
+    void set_function(void (Z::*function)()) { _function = function; }
+
+    /// Get arguments - don't need to do anything
+    virtual void get_arguments(std::string &line)
+    {
+        this->check_num_arguments(line,0);
+    }
+
+    /// Call the function
+    virtual void call_function()
+    {
+        std::cout << "\tCalling function with no arguments..." << std::flush;
+        (this->_object.get()->*_function)();
+        std::cout << "done." << std::endl;
+    }
+
+    /// Print number of arguments expected
+    virtual void print_num_arguments_expected() { std::cout << "\tNumber arguments expected: 0\n"; }
+
+
+protected:
+
+    /// Function
+    void (Z::*_function)();
+};
+
+/// Abstract base class for a parser key with 1 argument
+template<class Z, class A>
+class SIRFRegParserKey1Arg : public SIRFRegParserKeyBase<Z>
+{
+public:
+
+    /// Set function
+    void set_function(void (Z::*function)(A)) { _function = function; }
+
+    /// Get arguments
+    virtual void get_arguments(std::string &line)
+    {
+        this->check_num_arguments(line, 1);
+
+        // Set the argument
+        std::cout << "\tSetting argument...\n";
+        this->get_argument(line, 0, _arg1);
+        std::cout << "\t\tExpected type: " << get_typeid(_arg1) << "\n";
+        std::cout << "\t\tValue:         " << _arg1 << "\n";
+    }
+
+    /// Call the function
+    virtual void call_function()
+    {
+        std::cout << "\tCalling function with one argument..." << std::flush;
+        (this->_object.get()->*_function)(_arg1);
+        std::cout << "done." << std::endl;
+    }
+
+    /// Print number of arguments expected
+    virtual void print_num_arguments_expected() { std::cout << "\tNumber arguments expected: 1 (" << get_typeid(_arg1) << ")\n"; }
+
+protected:
+
+    /// Function
+    void (Z::*_function)(A);
+    /// Argument
+    A _arg1;
+};
+
+/// Abstract base class for a parser key with 2 arguments
+template<class Z, class A, class B>
+class SIRFRegParserKey2Arg : public SIRFRegParserKeyBase<Z>
+{
+public:
+
+    /// Set function
+    void set_function(void (Z::*function)(A,B)) { _function = function; }
+
+    /// Get arguments
+    virtual void get_arguments(std::string &line)
```

Comment:
```suggestion
    virtual void get_arguments(const std::string &line)
```
`static` or `const` method?